### ISSUE TITLE
Add tombstone handling in listing manifest (map phase)

### ISSUE THREAD

**Issue by shino** [basho/riak_cs#524]
Tombstones should be filtered out in a map phase of listing manifest.
https://github.com/basho/riak_cs/blob/release/1.3/src/riak_cs_utils.erl#L472

When there are tombstones, warning log is output to riak console.log:

```
2013-04-07 02:02:09.212 [warning] <0.13609.0>@riak_cs_utils:map_keys_and_manifests:483 Riak CS object list map failed: error:badarg
```

In case of Riak CS 1.2.2 and Riak 1.2.1, an error occurs and listing manifests fails
if there are tombstones in the minifest bucket.

CS's error log:

```
2013-04-07 21:11:47.797 [error] <0.524.0> webmachine error: path="/test/"
{error,{error,function_clause,[{riak_cs_s3_response,api_error,[<<"{\"phase\":0,\"error\":\"undef\",\"input\":\"{ok,{r_object,<<48,111,58,9,143,107,205,70,33,211,115,202
,222,78,131,38,39,180,246>>,<<\\\"2\\\">>,[{r_content,{dict,4,16,16,8,80,48,{[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[]},{{[],[],[],[],[],[],[],[],[],[],[[<<\\\"X-
Riak-VTag\\\">>,49,121,69,119,105,75,122,66,122,75,119,88,55,68,108,54,67,87,88,113,112,57]],[[<<\\\"index\\\">>]],[[<<\\\"X-Riak-Deleted\\\">>,116,114,117,101]],[[<<\\
\"X-Riak-Last-Modified\\\">>|{1365,336705,72013}]],[],[]}}},<<>>}],[{<<35,9,254,249,81,97,95,152>>,{5,63532555905}}],{dict,1,16,16,8,...},...},...}\",\"type\":\"error\"
,\"stack\":\"[{lager,dispatch_log,[warning,riak_cs_utils,map_keys_and_manifests,342,<0.1993.0>,[{module,riak_cs_utils},{function,map_keys_and_manifests},{line,342},{pid
,\\\"<0.1993.0>\\\"}],\\\"Riak CS object list map failed: ~p:~p\\\",[error,badarg],4096],[]},{riak_cs_utils,map_keys_and_manifests,3,[{file,\\\"src/riak_cs_utils.erl\\\
"},{line,342}]},{riak_kv_mrc_map,map,3,[{file,\\\"src/riak_kv_mrc_map.erl\\\"},{line,164}]},{riak_kv_mrc_map,process,3,[{file,\\\"src/riak_kv_mrc_map.erl\\\"},{line,140
}]},{riak_pipe_vnode_worker,process_input,...},...]\"}">>,{wm_reqdata,'GET',http,{1,1},"
```

The reason is `lager:dispatch_log` 's arity is changed from 8 (version 1.2.0)
to 9 (version 1.2.1).

**Comment by shino** [comment]
Quick fix on 1.2.2 : https://github.com/basho/riak_cs/compare/support;filter-tombstone#L0R332

**Comment by kuenishi** [comment]
Closing as #864 and #865 merged.
